extensions/s3/src/main/java/io/deephaven/extensions/s3/S3Instructions.java

```diff
@@ -135,6 +144,48 @@ public int numConcurrentWriteParts() {
         return DEFAULT_NUM_CONCURRENT_WRITE_PARTS;
     }
 
+    /**
+     * The default profile name used for configuring the default region, credentials, etc., when reading or writing to
+     * S3. If not provided, the AWS SDK picks the profile name from the 'aws.profile' system property, the "AWS_PROFILE"
+     * environment variable, or defaults to "default".
+     *
+     * @see ClientOverrideConfiguration.Builder#defaultProfileName(String)
+     */
+    public abstract Optional<String> profileName();
+
+    /**
+     * The path to the configuration file to use for configuring the default region, credentials, etc. when reading or
+     * writing to S3. If not provided, the AWS SDK picks the configuration file from the 'aws.configFile' system
+     * property, the "AWS_CONFIG_FILE" environment variable, or defaults to "{user.home}/.aws/config".
+     *
+     * @see ClientOverrideConfiguration.Builder#defaultProfileFile(ProfileFile)
+     */
+    public abstract Optional<Path> configFilePath();
+
+    /**
+     * The path to the credentials file to use for configuring the default region, credentials, etc. when reading or
+     * writing to S3. If not provided, the AWS SDK picks the credentials file from the 'aws.credentialsFile' system
+     * property, the "AWS_CREDENTIALS_FILE" environment variable, or defaults to "{user.home}/.aws/credentials".
+     *
+     * @see ClientOverrideConfiguration.Builder#defaultProfileFile(ProfileFile)
+     */
+    public abstract Optional<Path> credentialsFilePath();
+
+    /**
+     * The aggregated profile file that combines the configuration and credentials files.
+     */
+    final Optional<ProfileFile> aggregatedProfileFile() {
```

Comment:
`@Lazy` is probably better; `Optional` will probably be ok, but if that doesn't work with Lazy, you can probably use `@Nullable @Lazy`.